src/Bladeburner/ui/SkillElem.tsx

```diff
@@ -18,8 +18,14 @@ export function SkillElem({ skill, bladeburner, onUpgrade }: SkillElemProps): Re
   const skillName = skill.name;
   const skillLevel = bladeburner.getSkillLevel(skillName);
   const pointCost = useMemo(() => skill.calculateCost(skillLevel), [skill, skillLevel]);
-
-  const canLevel = bladeburner.skillPoints >= pointCost;
+  // No need to support "+1" button when the skill level reaches Number.MAX_SAFE_INTEGER.
+  const isSupported = skillLevel < Number.MAX_SAFE_INTEGER;
+  // Use skill.canUpgrade() instead of reimplementing all conditional checks.
+  const canLevel = isSupported ? skill.canUpgrade(bladeburner, 1).available ?? false : false;
```

Comment:
I really wanted to get rid of at least one of those falses. This is what I came up with:
```suggestion
  const canLevel = isSupported && skill.canUpgrade(bladeburner, 1).available;
```
I'm not sure it's actually any better, though.